### PR TITLE
Graphical Hangman + bug fixes 

### LINKED ISSUE #4: Leave game
As a user, I should be able to exit the game at any given time.

### LINKED ISSUE #7: Graphical hangman
As a user, I should be able to see a graphical representation of the hangman whilst playing the game. If I guess incorrectly, then the hangman should update, if I guess correctly then it should remain the same.

### LINKED ISSUE #19: Incomplete Game Ending: Final Guessed Word Not Printed
We should show the completed word at the end at the least so that users can see their correctly guessed word!

OOP ending:

```
It is a city with 9 letters! 
Word: A M S T E R _ A M 
Remaining lives: 6
Letters guessed: {'A', 'O', 'L', 'R', 'S', 'T', 'W', 'E', 'M'}
Please guess a letter: d
This letter is in the word!
Yay! You won the game!
Enter Y to replay the game or any other key to exit:
```

Current Ending:
````
Please guess a letter: a

This letter is in the word!
Word: P E T U N I A
Congratulations! You won the game!
Enter Y to replay the game. Enter any other key to exit.
````

### LINKED ISSUE #24: Game feels slow when playing
The game feels very slow because of the 1-second sleep delay that has been added throughout the game. This was done to make up for the "clear terminal screen" after letter guesses during the game.

Refactoring the code a bit will help get past the need for the sleep timer.

### PR DESCRIPTION
**Pull request by shaw-rohit**
A graphical stickman hangman has been added to the interface which updates based on lives remaining. Unfortunately, it's handicapped with 3 limbs because issue #19 still persists.

The best solution I can think of: Putting the print statements for
1. Graphical hangman
2. Hint
3. Health
4. Guessed letters

in a method and calling it in the while loop as well as won/lost conditions.

**Edit Apr 27:**
1. The proposed solution mentioned above has been implemented and the Hangman now has 4 limbs! (Fix #7)
2. The solution also fixed the incomplete game issue we were facing. (Fix #19)
3. The sleep timer has been adjusted from 1 to 0.5, making the game feel 2x faster and responsive. (Fix #24)
4. All print statements are now using interface.clear() method instead of "\n" for line breaks, as discussed during our call on Thursday.
5. Ctrl + C to exit has been added to the game instructions. (Fix #4)

### CODE REVIEW

**Comment by Pungent-Paws** [comment]
aww poor hangman :(  Yes, I think your proposed solution would work, imo we can go ahead with that

**Comment by shrutirao94** [comment]
Great!! Hangman works as you mentioned (with 3 limbs!)

File: hangman/interface.py [hunk 1]
```diff
@@ -78,22 +179,35 @@ def write_already_guessed(self):
...
     def write_won(self, word):
-        print("\nThe word: {0}\nYou guessed the word and won the game! Congratulations!".format(word))
+        self.clear()
+        print("You guessed the word and won the game! Congratulations!".format(word))
 
     def write_lost(self, word):
+        self.clear()
         print("Sorry! You ran out of health! The answer is {0}".format(word))
 
     def clear(self):
```
Comment (arumoy-shome) [inline hangman/interface.py:204]: probably rename this to `newlines`?

File: hangman/interface.py [hunk 2]
```diff
@@ -43,32 +49,127 @@ def write_instructions(self):
...
 
             if choice == "X":
                 input_flag = 1
 
             else:
-                print("  Invalid input. Please try again.\n")
+                self.clear()
+                print("Invalid input. Please try again.")
+
+    def draw_hangman(self, health):
```
Comment (arumoy-shome) [inline hangman/interface.py:72]: this can be a staticmethod since it does not need `self`

Comment (arumoy-shome) [reply hangman/interface.py:72]: come to think of it, all methods in Interface should/can be static 😅, perhaps we hold off on this change and refactor the code in a new PR (low priority).

File: hangman/interface.py [hunk 3]
```diff
@@ -43,32 +49,127 @@ def write_instructions(self):
...
+            print("   |      O  ")
+            print("   |     \|/ ")
+            print("   |      |  ")
+            print("   |     / \ ")
+            print("___|____     ")
 
     def read_replay(self):
         return input("Enter Y to replay the game or any other key to exit: ")
 
-    def write_credits(self):
```
Comment (arumoy-shome) [inline hangman/interface.py:63]: 🔥

File: hangman/interface.py [hunk 4]
```diff
@@ -43,32 +49,127 @@ def write_instructions(self):
             "3. If the letter you guessed doesn't belong to the word, an appendage to the stickman hangman will be drawn and you will lose one unit of life."
         )
         print(
-            "4. You have 9 tries (lives) to guess the word. You lose the game when you run out of lives and the character is hung.\n\n"
+            "4. You have 9 tries (lives) to guess the word. You lose the game when you run out of lives and the character is hung."
+        )
+        self.clear()
+        print(
```
Comment (arumoy-shome) [inline hangman/interface.py:55]: nitpick, but can we have everything on the same line?

File: hangman/game.py [hunk 5]
```diff
@@ -26,17 +26,13 @@ def create_hist(self):
...
-            self.interface.write_guessed_letters(self.player.guesses)
+            self.print_graphics()
             guess = self.interface.read_player_guess()
 
-            if(len(guess) == 1 and guess.isalpha()):
+            if len(guess) == 1 and guess.isalpha():
                 if guess in self.player.guesses:
                     self.interface.write_already_guessed()
-                    time.sleep(1)
+                    time.sleep(0.5)
```
[the remark this answers is not in the document]
Comment (shaw-rohit) [reply hangman/game.py:35]: > do we need this inside each if-else case?

Good catch! Moving the sleep timer to the end of the while loop 😁

File: hangman/interface.py (hunk 4 above)
Comment (shaw-rohit) [reply hangman/interface.py:55]: > nitpick, but can we have everything on the same line?

Do you mean like this:
`print("4. You have 9 tries (lives) to guess the word. You lose the game when you run out of lives and the character is hung.")`

File: .gitignore [hunk 6]
```diff
@@ -1,7 +1,7 @@
 # Byte-compiled
 .cache
 .pytest_cache/
-__pycache__/
+**/*/__pycache__/
```
[the remark this answers is not in the document]
Comment (shaw-rohit) [reply .gitignore:4]: Yep, will rebase off of master

File: hangman/interface.py (hunk 4 above)
Comment (arumoy-shome) [reply hangman/interface.py:55]: yup exactly!

File: hangman/game.py [hunk 7]
```diff
@@ -41,22 +40,23 @@ def start(self):
...
                     self.sethint(guess)
                     if "_" not in self.hint:
                         self.guessed = True
-                    time.sleep(0.5)
+
             else:
                 self.interface.write_invalid_guess()
-                time.sleep(0.5)
+                
+            time.sleep(0.5)
```
Comment (arumoy-shome) [inline hangman/game.py:54]: 👍 much cleaner!

File: hangman/interface.py (hunk 4 above)
Comment (shaw-rohit) [reply hangman/interface.py:55]: I blame the `black `formatter for this

**Pull request merged**